src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -64,11 +65,14 @@ public static string CleanUriSegment(string absolutePath)
                 }
 
                 // replace path segments that look like numbers or guid
-                segment = long.TryParse(segment, out _) ||
-                    Guid.TryParseExact(segment, "N", out _) ||
-                    Guid.TryParseExact(segment, "D", out _)
-                        ? UrlIdPlaceholder
-                        : segment;
+                // GUID format N "d85b1407351d4694939203acc5870eb1" length: 32
+                // GUID format D "d85b1407-351d-4694-9392-03acc5870eb1" length: 36 with dashses in 8, 13, 18 and 23 indexes.
+                if (long.TryParse(segment, out _) ||
+                    (segment.Length == 32 && Guid.TryParseExact(segment, "N", out _)) ||
+                    (segment.Length == 36 && segment[8] == '-' && segment[13] == '-' && segment[18] == '-' && segment[23] == '-' && Guid.TryParseExact(segment, "D", out _)))
```

Comment:
Do you think maybe only checking `Length == 36` is enough here? Or just the first dash instead of all 4? My (small) concern here is that when the GUID has format `"D"`, we're making things slower by adding 5 additional comparisons.

Some benchmarks would be nice, if possible 😄

Reply:
Benchmarks are below :)

Reply:
Thanks!